jina/peapods/pods/__init__.py

```diff
@@ -20,7 +20,63 @@
 from ...helper import get_public_ip, get_internal_ip, random_identity
 
 
-class BasePod(ExitStack):
+class ExitFIFO(ExitStack):
+    """
+    ExitFIFO changes the exiting order of exitStack to turn it into FIFO.
+    """
+
+    def __exit__(self, *exc_details):
+        received_exc = exc_details[0] is not None
+
+        # We manipulate the exception state so it behaves as though
+        # we were actually nesting multiple with statements
+        frame_exc = sys.exc_info()[1]
+
+        def _fix_exception_context(new_exc, old_exc):
+            # Context may not be correct, so find the end of the chain
+            while 1:
```

Comment:
Please do `while True:`

Reply:
Wont touch code here since is copy pasted from contextlib